src/persist/src/operators/stream.rs

```diff
@@ -288,68 +270,46 @@ where
                     return;
                 }
 
-                // Only try and seal if we have seen some data. Otherwise, we wouldn't have
-                // a capability that allows us to emit errors.
-                if let Some(err_cap) = capabilities.get(0) {
-                    for frontier_element in new_input_frontier.iter() {
-                        // Only seal if this element of the new input frontier truly
-                        // represents progress. With Antichain<u64>, this will always be
-                        // the case, but antichains of types with a different partial order
-                        // can have frontier progress and have some elements that don't
-                        // represent progress.
-                        if !input_frontier.less_than(frontier_element) {
-                            continue;
-                        }
+                for frontier_element in new_input_frontier.iter() {
+                    // Only seal if this element of the new input frontier truly
+                    // represents progress. With Antichain<u64>, this will always be
+                    // the case, but antichains of types with a different partial order
+                    // can have frontier progress and have some elements that don't
+                    // represent progress.
+                    if !input_frontier.less_than(frontier_element) {
+                        continue;
+                    }
 
-                        log::trace!("Sealing {} up to {}", &operator_name, frontier_element);
+                    log::trace!("Sealing {} up to {}", &operator_name, frontier_element);
 
-                        let result = write.seal(*frontier_element).await;
-                        if let Err(e) = result {
-                            log::error!(
-                                "Error sealing {} up to {}: {:?}",
-                                &operator_name,
-                                frontier_element,
-                                e
-                            );
+                    if let Err(e) = write.seal(*frontier_element).await {
+                        // If we fail to seal, simply exit and try again the next
+                        // time the operator is scheduled.
```

Comment:
Do we control when the operator is scheduled again?

Reply:
No, we don't. but the operator will be rescheduled if the frontier advances again or if theres new data

Reply:
pushed a new commit lmk what you think

Reply:
That seems good, yes.